commit 8f3ac98fdf40ab22f7754a5ec35b91a6585014b5
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Mon Jan 23 22:21:11 2017 +0800

    Add support for Qt 5.8+
    
    private/qgenericunixthemes_p.h is now under QtThemeSupport, using
    theme_support-private instead of platformsupport-private fixes build
    with latest Qt.
    
    Same for qwindow_p.h which is now under QtGui.
    
    Other adjustments including fileIconPixmap => fileIcon and an additional
    import on QtWidgets/qtwidgetsglobal.h for Q_WIDGETS_EXPORT.
    
    Change-Id: I7d48c640b251167b32f8423424ce5d61124f6da7

diff --git a/platformplugin/qt5platform-plugin.pro b/platformplugin/qt5platform-plugin.pro
index a2e262c..f235789 100644
--- a/platformplugin/qt5platform-plugin.pro
+++ b/platformplugin/qt5platform-plugin.pro
@@ -11,8 +11,13 @@ PLUGIN_CLASS_NAME = DXcbIntegrationPlugin
 DESTDIR = $$_PRO_FILE_PWD_/../bin/plugins/platforms
 
 QT       += opengl x11extras
-QT       += core-private platformsupport-private #xcb_qpa_lib-private
-greaterThan(QT_MAJOR_VERSION, 4): QT += widgets widgets-private
+QT       += core-private #xcb_qpa_lib-private
+greaterThan(QT_MAJOR_VERSION, 4) {
+  QT += widgets widgets-private
+  # Qt >= 5.8
+  greaterThan(QT_MAJOR_VERSION, 5)|greaterThan(QT_MINOR_VERSION, 7): QT += gui-private
+  else: QT += platformsupport-private
+}
 
 TEMPLATE = lib
 VERSION = $$QT_VERSION
diff --git a/platformplugin/utility_x11.cpp b/platformplugin/utility_x11.cpp
index 1c4ce6f..9c045dd 100644
--- a/platformplugin/utility_x11.cpp
+++ b/platformplugin/utility_x11.cpp
@@ -9,6 +9,10 @@
 #include <QDebug>
 #include <QtX11Extras/QX11Info>
 
+#if QT_VERSION >= 0x050800
+    #include <QtWidgets/qtwidgetsglobal.h>
+#endif
+
 #include <xcb/shape.h>
 
 #include <X11/cursorfont.h>
diff --git a/platformthemeplugin/qdeepintheme.cpp b/platformthemeplugin/qdeepintheme.cpp
index 004507f..2a25542 100644
--- a/platformthemeplugin/qdeepintheme.cpp
+++ b/platformthemeplugin/qdeepintheme.cpp
@@ -122,12 +122,21 @@ QPixmap QDeepinTheme::standardPixmap(QPlatformTheme::StandardPixmap sp, const QS
     return QGenericUnixTheme::standardPixmap(sp, size);
 }
 
+#if QT_VERSION >= 0x050800
+QIcon QDeepinTheme::fileIcon(const QFileInfo &fileInfo, QPlatformTheme::IconOptions iconOptions) const
+{
+    Q_UNUSED(iconOptions);
+
+    return XdgIcon::fromTheme(m_mimeDatabase.mimeTypeForFile(fileInfo).iconName());
+}
+#else
 QPixmap QDeepinTheme::fileIconPixmap(const QFileInfo &fileInfo, const QSizeF &size, QPlatformTheme::IconOptions iconOptions) const
 {
     Q_UNUSED(iconOptions);
 
     return XdgIcon::fromTheme(m_mimeDatabase.mimeTypeForFile(fileInfo).iconName()).pixmap(size.toSize());
 }
+#endif
 
 QVariant QDeepinTheme::themeHint(QPlatformTheme::ThemeHint hint) const
 {
diff --git a/platformthemeplugin/qdeepintheme.h b/platformthemeplugin/qdeepintheme.h
index c9f8e09..697bc71 100644
--- a/platformthemeplugin/qdeepintheme.h
+++ b/platformthemeplugin/qdeepintheme.h
@@ -18,8 +18,13 @@ public:
 
     QIconEngine *createIconEngine(const QString &iconName) const Q_DECL_OVERRIDE;
     QPixmap standardPixmap(StandardPixmap sp, const QSizeF &size) const Q_DECL_OVERRIDE;
+#if QT_VERSION >= 0x050800
+    QIcon fileIcon(const QFileInfo &fileInfo,
+                   QPlatformTheme::IconOptions iconOptions = 0) const Q_DECL_OVERRIDE;
+#else
     QPixmap fileIconPixmap(const QFileInfo &fileInfo, const QSizeF &size,
                            QPlatformTheme::IconOptions iconOptions = 0) const Q_DECL_OVERRIDE;
+#endif
 
     QVariant themeHint(ThemeHint hint) const Q_DECL_OVERRIDE;
 
diff --git a/platformthemeplugin/qt5deepintheme-plugin.pro b/platformthemeplugin/qt5deepintheme-plugin.pro
index c37c6b1..3b70761 100644
--- a/platformthemeplugin/qt5deepintheme-plugin.pro
+++ b/platformthemeplugin/qt5deepintheme-plugin.pro
@@ -5,8 +5,13 @@
 #-------------------------------------------------
 
 QT       += dbus x11extras
-QT       += core-private gui-private platformsupport-private
-greaterThan(QT_MAJOR_VERSION, 4): QT += widgets widgets-private
+QT       += core-private gui-private
+greaterThan(QT_MAJOR_VERSION, 4) {
+  QT += widgets widgets-private
+  # Qt >= 5.8
+  greaterThan(QT_MAJOR_VERSION, 5)|greaterThan(QT_MINOR_VERSION, 7): QT += theme_support-private
+  else: QT += platformsupport-private
+}
 
 TARGET = qdeepin
 TEMPLATE = lib
